scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingService.java

```diff
@@ -259,6 +267,46 @@ public void run() {
         return true;
     }
 
+    /**
+     * Create a new Credential object containing users' 3rd Party Credentials.
+     *
+     * @param creds credentials for specific user
+     * @return in case of success new object containing the 3rd party credentials used to create bindings
+     * at clean script
+     */
+    Credentials addThirdPartyCredentials(Credentials creds) throws KeyException, IllegalAccessException {
+        //retrieve scheduler key pair
+        String privateKeyPath = PASchedulerProperties.getAbsolutePath(PASchedulerProperties.SCHEDULER_AUTH_PRIVKEY_PATH.getValueAsString());
+        String publicKeyPath = PASchedulerProperties.getAbsolutePath(PASchedulerProperties.SCHEDULER_AUTH_PUBKEY_PATH.getValueAsString());
+
+        //get keys from task
+        PrivateKey privateKey = Credentials.getPrivateKey(privateKeyPath);
+        PublicKey publicKey = Credentials.getPublicKey(publicKeyPath);
+
+        //retrieve the current creData from task
+        CredData credData = creds.decrypt(privateKey);
+
+        //retrive database to get third party credentials from
+        SchedulerDBManager dbManager = getInfrastructure().getDBManager();
+        if (dbManager != null) {
+            Map<String, HybridEncryptedData> thirdPartyCredentials = null;
+            thirdPartyCredentials = dbManager.thirdPartyCredentialsMap(credData.getLogin());
+            if (thirdPartyCredentials == null) {
+                logger.error("Failed to retrieve Third Party Credentials!");
+                throw new KeyException("Failed to retrieve thirdPartyCredentials!");
+            } else {
+                //cycle third party credentials, add one-by-one to the decrypter
+                for (Map.Entry<String, HybridEncryptedData> thirdPartyCredential : thirdPartyCredentials.entrySet()) {
+                    String decryptedValue = HybridEncryptionUtil.decryptString(thirdPartyCredential.getValue(),
+                                                                               privateKey);
+                    credData.addThirdPartyCredential(thirdPartyCredential.getKey(), decryptedValue);
+                }
+            }
+        }
+        Credentials nodeEncryptedUserCredentials = Credentials.createCredentials(credData, publicKey);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "nodeEncryptedUserCredentials". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1488)

Reply:
like Sonar said, return Credentials.createCredentials(credData, publicKey);